src/Raven.Server/Smuggler/Documents/ShardedDatabaseSmuggler.cs

```diff
@@ -248,16 +263,43 @@ await foreach (var kvp in _source.GetCompareExchangeValuesAsync())
                 }
 
             }
+
             return result.CompareExchange;
         }
 
         protected override async Task<SmugglerProgressBase.Counts> ProcessCompareExchangeTombstonesAsync(SmugglerResult result)
         {
-            await using (var destinationActions = _destination.CompareExchange(_context))
+            await using (var destinationActions = _destination.CompareExchangeTombstones(_context))
+            await using (var actions = new DatabaseCompareExchangeActions(_server, _databaseRecord, _context, new CancellationToken()))
             {
                 await foreach (var kvp in _source.GetCompareExchangeTombstonesAsync())
                 {
-                    await destinationActions.WriteTombstoneKeyAsync(kvp.Key.Key);
+                    _token.ThrowIfCancellationRequested();
+                    result.CompareExchangeTombstones.ReadCount++;
+
+                    if (kvp.Equals(default))
+                    {
+                        result.CompareExchangeTombstones.ErroredCount++;
+                        continue;
+                    }
+
+                    try
+                    {
+                        if (ClusterTransactionCommand.IsAtomicGuardKey(kvp.Key.Key, out _))
+                        {
+                            DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Aviv, DevelopmentHelper.Severity.Normal, "handle atomic guard tombstones import");
```

Comment:
Do we have a test to check this scenario? if not, let's create an issue and move on

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-19201/Sharding-Import-handle-atomic-guard-tombstones